pkg/nirmata/resource_cluster_addons.go

```diff
@@ -0,0 +1,173 @@
+package nirmata
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/nirmata/go-client/pkg/client"
+)
+
+func resoureClusterAddon() *schema.Resource {
+	return &schema.Resource{
+
+		Create: resourceClusterAddonCreate,
+		Read:   resourceClusterAddonRead,
+		Update: resourceClusterAddonUpdate,
+		Delete: resourceClusterAddonDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateName,
+			},
+			"cluster": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"application": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"environment": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"catalog": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"namespace": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"channel": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"service_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"service_port": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"service_scheme": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+func resourceClusterAddonCreate(d *schema.ResourceData, meta interface{}) error {
+	apiClient := meta.(client.Client)
+	name := d.Get("name").(string)
+	clusterNameOrID := d.Get("cluster").(string)
+	namespace := d.Get("namespace").(string)
+	application := d.Get("application").(string)
+	channel := d.Get("channel").(string)
+	catalog := d.Get("catalog").(string)
+	environment := d.Get("environment").(string)
+	labels := d.Get("labels")
+
+	if namespace == "" {
+		namespace = application
+	}
+	if environment == "" {
+		environment = application + "-" + clusterNameOrID
```

Comment:
if an ID is provided, instead of the name, should we shorten the env name to have a 6 digit unique suffix? or we can query the cluster name, using the ID, and then append the cluster name.

Reply:
Query cluster name and append.